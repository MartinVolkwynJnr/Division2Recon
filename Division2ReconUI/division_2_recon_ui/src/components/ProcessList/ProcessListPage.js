import React, { useState, useEffect } from "react";
import ProcessList from "./ProcessList";
import TextInput from "../common/TextInput";
import {  getProcesses, SearchByCustomerName, SearchBySensorData } from "../../api/processesApi";

function ProcessListPage() {
  const [processes, setProcesses] = useState([]);
  const [searchString, setSearchString] = useState("");

  useEffect(() => {
    getProcesses().then(_processes => {
      setProcesses(_processes);
    });
  },[])

  function handleSearch(event) {
      setSearchString(event.target.value);
  }

  function handleClearSearch(event) {
    setSearchString("");
    getProcesses().then(_processes => {
      setProcesses(_processes);
    });
}

  function searchByCustomerName(){
    console.log("searchString searchByCustomerName",searchString);
    if(searchString !== ""){
      SearchByCustomerName(searchString).then(_processes => setProcesses(_processes));
    }
    else{
      getProcesses().then(_processes => setProcesses(_processes))
    }
    
  }

  function searchBySensorData(event){
    console.log("searchString SearchBySensorData",searchString);
    if(searchString !== ""){
      SearchBySensorData(searchString).then(_processes => setProcesses(_processes));
    }
    else{
      getProcesses().then(_processes => setProcesses(_processes))
    }
  }

  return (
    <div>
      <TextInput
        name="searchInput"
        label="Search"
        value={searchString}
        onInput={handleSearch}
        error={""}
      />
      <button onClick={searchByCustomerName}>Search CustomerName</button>
      <button onClick={searchBySensorData}>Search SensorData</button>
      <button onClick={handleClearSearch}>Clear Search</button>
      <ProcessList
        processes={processes}
      />
    </div>
  );
}

export default ProcessListPage
