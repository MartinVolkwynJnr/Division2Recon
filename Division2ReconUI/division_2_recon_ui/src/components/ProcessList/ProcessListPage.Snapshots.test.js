import React from "react";
import ProcessListPage from "./ProcessListPage";
import renderer from "react-test-renderer";
import {render} from "@testing-library/react";

it("check general display of snapshot change/update", () => {
    const tree = renderer.create(
        <ProcessListPage />
    );
    expect(tree).toMatchSnapshot();
})

it("should render a button with text 'Search CustomerName'", () => {
    const {getByText} = render(<ProcessListPage />);
    getByText("Search CustomerName");
})

it("should render a button with text 'Search SensorData'", () => {
    const {getByText} = render(<ProcessListPage />);
    getByText("Search SensorData");
})