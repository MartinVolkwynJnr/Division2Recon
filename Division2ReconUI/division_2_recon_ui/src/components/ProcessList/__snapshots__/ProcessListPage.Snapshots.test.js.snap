// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check general display of snapshot change/update 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      htmlFor="searchInput"
    >
      Search
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="searchInput"
        onInput={[Function]}
        type="text"
        value=""
      />
      
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    Search CustomerName
  </button>
  <button
    onClick={[Function]}
  >
    Search SensorData
  </button>
  <div>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            CustomerId
          </th>
          <th>
            CustomerName
          </th>
          <th>
            MachineNr
          </th>
          <th>
            MachineId
          </th>
          <th>
            MachineTypeSerial
          </th>
          <th>
            Process
          </th>
          <th>
            ProcessTime
          </th>
          <th>
            SensorData
          </th>
          <th>
            OnlineFrom
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
